@xen-orchestra/lite/src/components/infra/InfraHostItem.vue

```diff
@@ -58,6 +62,14 @@ const isCurrentHost = computed(
   () => props.hostOpaqueRef === uiStore.currentHostOpaqueRef
 );
 const [isExpanded, toggle] = useToggle(true);
+
+const getVmCount = computed(() => {
+  const { recordsByHostRef } = useVmCollection();
+  const vms = recordsByHostRef.value.get(
+    props.hostOpaqueRef ?? ("OpaqueRef:NULL" as XenApiHost["$ref"])
+  );
```

Comment:
Why did you add ` ?? ("OpaqueRef:NULL" as XenApiHost["$ref"])`
```suggestion
  const vms = recordsByHostRef.value.get(props.hostOpaqueRef);
```
`props.hostOpaqueRef` is not supposed to be `undefined` or `null`.
```ts
...
const props = defineProps<{
  hostOpaqueRef: XenApiHost["$ref"];
}>();
...
```

Reply:
I used vms const from InfraVmList.vue, isn't it appropriate?

Reply:
The [nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) is a tool in JavaScript designed to provide a fallback value when the left-hand side of the expression is null or undefined. Its syntax is straightforward: `a ?? b` returns `a` if `a` is neither null nor undefined, otherwise, it returns `b`.

In `InfraHostItem.vue`, `props.hostOpaqueRef` is defined with a guarantee of non-nullability. This means that in the context of this code, `props.hostOpaqueRef` cannot be `undefined`, so using the `nullish coalescing operator (??)` is unnecessary.
```ts
...
const props = defineProps<{
  hostOpaqueRef: XenApiHost["$ref"];
}>();
...
```
In `InfraVmList.vue`, `props.hostOpaqueRef` is declared as potentially `undefined`, then the `??` operator is justified to provide a default value.
```ts
// InfraVmList.vue
...
const props = defineProps<{
  hostOpaqueRef?: XenApiHost["$ref"];
}>();
...
```
the `?` after the `hostOpaqueRef` mean the value can be `undefined`.